overlord/state/taskrunner.go

```diff
@@ -468,6 +511,21 @@ ConsiderTasks:
 			}
 		}
 
+		unblocked, minCheckAfter, err := r.checkBlockingConditions(t)
+		if err != nil {
+			return err
+		}
+
+		if !unblocked {
+			// check again before the requested time, if any was set
+			if minCheckAfter > 0 &&
+				(nextTaskTime.IsZero() || nextTaskTime.After(timeNow().Add(minCheckAfter))) {
+				r.state.EnsureBefore(minCheckAfter)
+			}
```

Comment:
shouldn't we instead rename nextTaskTime maybe and set it instead of having an EnsureBefore here and below?